server.py

```diff
@@ -0,0 +1,37 @@
+from gevent.server import StreamServer
+from gevent.monkey import patch_all
+from echo_server import response_ok, parse_request, resolve_uri
+
+def handle(conn, addr):
+    bsize = 32
+    try:
+        while True:
```

Comment:
This loop does you no good, and actually hurts you.  If you were to run this, each request handled would end with a socket error as the loop came around a second time, attempted to `recv` from a closed socket on line 12, and borked.  

When you plug a handler like this into a server provided by `gevent`, you are really only responsible for handling the one connection even given to you by the server.  Each individual connection is handled in a separate execution of your handler, so you don't need to deal with repeating the process yourself.